beeai/agents/backport_agent.py

```diff
@@ -22,45 +20,41 @@
 from beeai_framework.tools import Tool
 from beeai_framework.tools.search.duckduckgo import DuckDuckGoSearchTool
 from beeai_framework.tools.think import ThinkTool
+from beeai_framework.workflows import Workflow
 
+import tasks
+from constants import COMMIT_PREFIX
+from observability import setup_observability
+from tools.commands import RunShellCommandTool
 from tools.specfile import AddChangelogEntryTool, BumpReleaseTool
 from tools.text import CreateTool, InsertTool, StrReplaceTool, ViewTool
 from tools.wicked_git import GitLogSearchTool, GitPatchCreationTool
-from constants import COMMIT_PREFIX, BRANCH_PREFIX
-from observability import setup_observability
-from tools.commands import RunShellCommandTool
 from triage_agent import BackportData, ErrorData
-from utils import get_agent_execution_config, mcp_tools, redis_client, get_git_finalization_steps
+from utils import check_subprocess, get_agent_execution_config, mcp_tools, redis_client
 
 logger = logging.getLogger(__name__)
 
 
 class InputSchema(BaseModel):
+    local_clone: Path = Field(description="Path to the local clone of forked dist-git repository")
+    unpacked_sources: Path = Field(description="Path to the unpacked (using `centpkg prep`) sources")
     package: str = Field(description="Package to update")
+    dist_git_branch: str = Field(description="Git branch in dist-git to be updated")
     upstream_fix: str = Field(description="Link to an upstream fix for the issue")
     jira_issue: str = Field(description="Jira issue to reference as resolved")
     cve_id: str = Field(default="", description="CVE ID if the jira issue is a CVE")
-    dist_git_branch: str = Field(description="Git branch in dist-git to be updated")
-    git_repo_basepath: str = Field(
-        description="Base path for cloned git repos",
-        default=os.getenv("GIT_REPO_BASEPATH"),
-    )
-    unpacked_sources: str = Field(
-        description="Path to the unpacked (using `centpkg prep`) sources",
-        default="",
-    )
 
 
 class OutputSchema(BaseModel):
     success: bool = Field(description="Whether the backport was successfully completed")
     status: str = Field(description="Backport status")
-    mr_url: Optional[str] = Field(description="URL to the opened merge request")
-    error: Optional[str] = Field(description="Specific details about an error")
+    mr_url: str | None = Field(description="URL to the opened merge request")
+    error: str | None = Field(description="Specific details about an error")
 
 
 def render_prompt(input: InputSchema) -> str:
     template = (
-        'Work inside the repository cloned at "{{ git_repo_basepath }}/{{ package }}"\n'
+        'Work inside the repository cloned in "{{ local_clone }}", it is your current working directory\n'
```

Comment:
nice, love this change
I think it was annoying and unpredictable that the agents had to `cd /...` all the time